allennlp/commands/train.py

```diff
@@ -178,42 +193,47 @@ def create_serialization_dir(params: Params, serialization_dir: str, recover: bo
         If ``True``, we will try to recover from an existing serialization directory, and crash if
         the directory doesn't exist, or doesn't match the configuration we're given.
     """
+    if recover and force:
+        raise ConfigurationError("Illegal arguments: both force and recover are true.")
+
     if os.path.exists(serialization_dir) and os.listdir(serialization_dir):
-        if not recover:
+        if force:
```

Comment:
The level of nesting that this gets to is scary.  I also think you can avoid it - what if you just move the remove block up above, and leave the rest of the code unmodified?  That is, have something like:

```python
if recover and force:
    raise ...

if force:
    shutil.rmtree(serialization_dir)

...  # the rest of the code unchanged
```